tests/test_fetcher_ng.py

```diff
@@ -136,44 +139,44 @@ def test_response_read_timeout(self, mock_session_get):
     @patch.object(
         requests.Session, "get", side_effect=urllib3.exceptions.TimeoutError
     )
-    def test_session_get_timeout(self, mock_session_get):
+    def test_session_get_timeout(self, mock_session_get: Any) -> None:
         with self.assertRaises(exceptions.SlowRetrievalError):
             self.fetcher.fetch(self.url)
         mock_session_get.assert_called_once()
 
     # Simple bytes download
-    def test_download_bytes(self):
+    def test_download_bytes(self) -> None:
         data = self.fetcher.download_bytes(self.url, self.file_length)
         self.assertEqual(self.file_contents, data.decode("utf-8"))
 
     # Download file smaller than required max_length
-    def test_download_bytes_upper_length(self):
+    def test_download_bytes_upper_length(self) -> None:
         data = self.fetcher.download_bytes(self.url, self.file_length + 4)
         self.assertEqual(self.file_contents, data.decode("utf-8"))
 
     # Download a file bigger than expected
-    def test_download_bytes_length_mismatch(self):
+    def test_download_bytes_length_mismatch(self) -> None:
         with self.assertRaises(exceptions.DownloadLengthMismatchError):
             self.fetcher.download_bytes(self.url, self.file_length - 4)
 
     # Simple file download
-    def test_download_file(self):
+    def test_download_file(self) -> None:
         with self.fetcher.download_file(
             self.url, self.file_length
         ) as temp_file:
             temp_file.seek(0, io.SEEK_END)
             self.assertEqual(self.file_length, temp_file.tell())
 
     # Download file smaller than required max_length
-    def test_download_file_upper_length(self):
+    def test_download_file_upper_length(self) -> None:
         with self.fetcher.download_file(
             self.url, self.file_length + 4
         ) as temp_file:
             temp_file.seek(0, io.SEEK_END)
             self.assertEqual(self.file_length, temp_file.tell())
 
     # Download a file bigger than expected
-    def test_download_file_length_mismatch(self):
+    def test_download_file_length_mismatch(self) -> Iterator[Any]:
         with self.assertRaises(exceptions.DownloadLengthMismatchError):
             yield self.fetcher.download_file(self.url, self.file_length - 4)
```

Comment:
this has to be a bug: why would a test ever yield?

(not an important detail though: we expect a raise anyway)

Reply:
It is actually a way to force download_file to execute and raise the error since it is a context manager and returns Iterator[IO]. The other option would be probably:
```
with self.fetcher.download_file(self.url, self.file_length - 4) as f:
    pass
```

Reply:
I think it's more clear what happens the way it's now.
Maybe I can add a comment providing some information on why we yield here.
Do you agree?